src/librustc_mir/interpret/snapshot.rs

```diff
@@ -285,12 +286,24 @@ impl<'a, Ctx> Snapshot<'a, Ctx> for &'a Allocation
     type Item = AllocationSnapshot<'a>;
 
     fn snapshot(&self, ctx: &'a Ctx) -> Self::Item {
-        let Allocation { bytes, relocations, undef_mask, align, mutability, extra: () } = self;
+        let Allocation {
+            relocations,
+            size,
+            align,
+            mutability,
+            extra: (),
+            ..
```

Comment:
Same concern with the `..` pattern here.

Maybe add a comment at the `Allocation` type like
```
// NOTE: When adding new fields, make sure to adjust the StableHash impl in ... and the Snapshot impl in ...
```